carmajava/signal_plugin/src/main/java/gov/dot/fhwa/saxton/carma/signal_plugin/TrafficSignalPlugin.java

```diff
@@ -192,40 +196,80 @@ public void onInitialize() {
+
+                    response.setMessage(this.getVersionInfo().componentName() + " did not expect user input therefore ignoring the request.");
+                    response.setSuccess(false); //Set to fail to notify caller it wasn't expected.
+                    return;
                 }
-                if (request.getData()) { // If user acknowledged it is safe to continue
-                    log.info("User confirmed: All Clear");
-                    if (checkCurrentPhase(SignalPhase.GREEN)) { // If we are still in the green phase
-                        // Notify the arbitrator we need to replan to continue through the intersection
-                        log.info("Continuing at new green light");
-                        triggerNewPlan(true);
-                    } else {
-                        log.info("Light not green despite user confirmation");
-                        response.setMessage("The light is no longer green. Waiting till next green light.");
-                        response.setSuccess(false);
+
+                // If user acknowledged YES that it is safe to continue.
+                if (request.getData()) {
+
+                    log.info("User confirmed all clear.");
+                    log.info("GO_BUTTON_SRVS 3: awaitingUserConfirmation: " + awaitingUserConfirmation + "; awaitingUserInput:" + awaitingUserInput);
+
+                    if (checkCurrentPhase(SignalPhase.GREEN)) { // If still in the GREEN phase
+
+                        //Set false to acknowledge receipt
                         awaitingUserInput.set(false);
+                        awaitingUserConfirmation.set(false);
+
+                        // Notify the arbitrator we need to replan to continue through the intersection
+                        log.info("Current Phase is GREEN after user confirmed. Continuing through intersection and trigger new plan.");
+                        log.info("GO_BUTTON_SRVS 4: awaitingUserConfirmation: " + awaitingUserConfirmation + "; awaitingUserInput:" + awaitingUserInput);
+
+                        // Update vehicle position with most recent transform
+                        updateCurrentLocation(); //INSTEAD OF directly calling triggerNewPlan(true);
+
+                        response.setSuccess(true);
+                        return;
+
+                    } else {//Else no longer green.
+
+                        log.info("Current Phase is NO LONGER GREEN after user confirmed");
+                        log.info("GO_BUTTON_SRVS 5: awaitingUserConfirmation: " + awaitingUserConfirmation + "; awaitingUserInput:" + awaitingUserInput);
+
+                        response.setMessage("No longer green. Please wait to respond until the next green light."); //TODO: TEST
+                        response.setSuccess(true);  //Need to set to false in order to leave the popup open instead of success which the popup disappears.
```

Comment:
Comment says this should be false but it is being set to true

Reply:
please check latest code changes.